### PR TITLE
Pin gems and set some platforms as jruby

### PR DESCRIPTION
**Pull request by navarone-feekery**
In an effort to fix the renovate PRs, this PR tries to set the platform for specific gems to `:jruby`. This will stop the renovate bot from fetching these gems from the MRI ruby source.

I decided which gems to add the platform to by looking at all gems with the `-java` prefix in https://github.com/elastic/crawler/blob/main/Gemfile.lock

I also saw a good opportunity to alphabetize the gems and pin a version for every dependency (excluding test or dev dependencies).

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -173,8 +176,9 @@ DEPENDENCIES
   ruby-debug-ide
   simplecov
   simplecov-material
-  thread_safe (= 0.3.6)
-  tzinfo-data
+  strscan (~> 3.1.0)
```
Comment (artem-shelkovnikov) [inline Gemfile.lock:179]: What's up with strscan? Is it a indirect dependancy?

**Review by artem-shelkovnikov — APPROVED** [review]

File: Gemfile [hunk 2]
```diff
@@ -25,30 +22,34 @@ source 'https://rubygems.org'
...
   gem 'concurrent-ruby', '~> 1.1.4'
+  gem 'dry-cli', '~> 0.7.0'
   gem 'elasticsearch', '~> 8.13.0'
-  gem 'nokogiri', '= 1.13.10', require: false
-  gem 'json-schema'
-  gem 'activesupport', '= 6.1.7.7'
   gem 'jar-dependencies', '0.4.1'
+  gem 'json', '~> 2.7.2', platform: :jruby
+  gem 'json-schema', '~> 4.3.0'
+  gem 'nokogiri', '= 1.13.10', require: false, platform: :jruby
```
Comment (seanstory) [inline Gemfile:35]: why `require: false`? Don't we heavily need nokogiri in the crawler?

**Review by seanstory — APPROVED** [review]
one question, but assuming it's intentional, LGTM

File: Gemfile (hunk 2 above)
Comment (navarone-feekery) [reply Gemfile:35]: It was already `require: false` since ent-search days and I never questioned it. I think this would be safe to remove.

Comment (navarone-feekery) [reply Gemfile]: @seanstory I removed the `require: false` and also rearranged the jruby gems so it's easier to read.

**Review by seanstory — APPROVED** [review]

**Pull request merged**